Assignment 1/src/main/java/com/nuclei/assignment/service/itemparser/ItemParserImpl.java

```diff
@@ -0,0 +1,121 @@
+      throw new AttributeParseException(
+          String.format(ExceptionsConstantsUtils.INVALID_PRICE, price), exception);
+    }
+    return parsedPrice;
+  }
+  
+  @Override
+  public ItemType parseType(final String type) throws AttributeParseException {
+    validateData(type, "type");
+    final ItemType parsedItemType;
+    try {
+      parsedItemType = ItemType.valueOf(type.toUpperCase(Locale.ROOT));
+    } catch (IllegalArgumentException exception) {
+      logger.error(
+          String.format(ExceptionsConstantsUtils.INVALID_TYPE, type),
+          new Throwable().fillInStackTrace());
+      throw new AttributeParseException(
+          String.format(ExceptionsConstantsUtils.INVALID_TYPE, type), exception);
+    }
+    return parsedItemType;
+  }
+  
+  @Override
+  public double parseQuantity(final String quantity) throws AttributeParseException {
+    validateData(quantity, "quantity");
+    final double parsedQuantity;
+    try {
+      parsedQuantity = Double.parseDouble(quantity);
+      if (parsedQuantity < 0) {
+        logger.error(
+            String.format(ExceptionsConstantsUtils.NEGATIVE_QUANTITY, quantity),
+            new Throwable().fillInStackTrace());
+        throw new AttributeParseException(
+            String.format(ExceptionsConstantsUtils.NEGATIVE_QUANTITY, quantity));
+      }
+    } catch (NumberFormatException exception) {
+      logger.error(
+          String.format(ExceptionsConstantsUtils.INVALID_QUANTITY,
+          quantity), new Throwable().fillInStackTrace());
+      throw new AttributeParseException(
+          String.format(ExceptionsConstantsUtils.INVALID_QUANTITY, quantity), exception);
+    }
+    return parsedQuantity;
+  }
+  
+  private void validateData(String data, String checkedOnAttributeName)
+      throws AttributeParseException {
+    checkDataIsNotNull(data, checkedOnAttributeName);
+    checkDataIsNotFlag(data);
+  }
+  
+  private void checkDataIsNotFlag(String data) throws AttributeParseException {
+    if (data.equals(FlagsConstantsUtils.NAME_FLAG)
+        || data.equals(FlagsConstantsUtils.TYPE_FLAG)
+        || data.equals(FlagsConstantsUtils.QUANTITY_FLAG)
+        || data.equals(FlagsConstantsUtils.PRICE_FLAG)) {
+      logger.error(String.format(ExceptionsConstantsUtils.INVALID_INPUT_DATA_NOT_FLAG,
+          data), new Throwable().fillInStackTrace());
+      throw new AttributeParseException(
+          String.format(ExceptionsConstantsUtils.INVALID_INPUT_DATA_NOT_FLAG, data));
+    }
+  }
+  
+  private void checkDataIsNotNull(String data, String checkedOnAttributeName)
+      throws AttributeParseException {
+    if (data == null) {
+      logger.error(String.format(ExceptionsConstantsUtils.INVALID_INPUT_DATA_NOT_NULL,
+          checkedOnAttributeName), new Throwable().fillInStackTrace());
+      throw new AttributeParseException(
+          String.format(ExceptionsConstantsUtils.INVALID_INPUT_DATA_NOT_NULL,
+          checkedOnAttributeName));
+    }
+  }
```

Comment:
remove this after adding the validation for value corresponding to flag in validateItemProperties.

Reply:
It' still there as removing it will fail certain cases